src/DynamicRupture/FrictionLaws/ThermalPressurization/ThermalPressurization.h

```diff
@@ -0,0 +1,155 @@
+    real factor = 1 / std::sqrt(2.0 * M_PI);
+
+    for (size_t i = 0; i < N; ++i) {
+      real heatGeneration = std::exp(-0.5 * misc::power<2>(localGridPoints[i]));
+      values[i] = factor * heatGeneration;
+    }
+  }
+  real const& operator[](size_t i) const { return values[i]; };
+
+  private:
+  real values[N]{0};
+};
+
+/**
+ * We follow Noda&Lapusta (2010).
+ * Define: \f$p, T\f$ pressure and temperature, \f$\Pi, \Theta\f$ fourier transform of pressure and
+ * temperature respectively, \f$\Sigma = \Pi + \Lambda^\prime \Theta\f$. We solve equations (6) and
+ * (7) with the method from equation(10).
+ * \f[\begin{aligned}\text{Equation 6:} && \frac{\partial \Theta}{\partial t} =& -l^2 \alpha_{th}
+ * \Theta + \frac{\Omega}{\rho c}\\ \text{Equation 7:} && \frac{\partial \Sigma}{\partial t} =& -l^2
+ * \alpha_{hy} \Theta + (\Lambda + \Lambda^\prime) \frac{\Omega}{\rho c}\\\end{aligned}\f] with \f$
+ * \Omega = \tau V \frac{\exp(-l^2 w^2 / 2) }{\sqrt{2\pi}}\f$. We define \f$\hat{l} = lw \in
+ * [0,10]\f$ (see comment in [15]). Now, we can apply the solution procedure from equation (10) to
+ * get:
+ * \f[ \begin{aligned}\Theta(t+\Delta t) &= \frac{\Omega}{\rho c l^2 \alpha_{th}} \left[1 -
+ * \exp\left(-l^2 \alpha_{th} \Delta t\right)\right] + \Theta(t)\exp(-l^2\alpha_{th} \Delta t)\\
+ * &= \frac{\tau V}{\sqrt{2\pi}\rho c \left(\hat{l}/w\right)^2 \alpha_{th}} \exp(-\hat{l}^2/2)
+ * \left[1 - \exp\left(-\left(\hat{l}/w\right)^2 \alpha_{th} \Delta t\right)\right]
+ * + \Theta(t)\exp\left(-\left(\hat{l}/w\right)^2\alpha_{hy} \Delta t\right)\\\end{aligned}\f]
+ * and
+ * \f[ \begin{aligned}\Sigma(t+\Delta t) &= \frac{(\Lambda + \Lambda^\prime)\Omega}{\rho c l^2
+ * \alpha_{hy}} \left[1 - \exp\left(-l^2 \alpha_{hy} \Delta t\right)\right] +
+ * \Theta(t)\exp(-l^2\alpha_{hy} \Delta t)\\
+ * &= \frac{(\Lambda + \Lambda^\prime)\tau V}{\sqrt{2\pi}\rho c \left(\hat{l}/w\right)^2
+ * \alpha_{hy}} \exp(-\hat{l}^2/2) \left[1 - \exp\left(-\left(\hat{l}/w\right)^2 \alpha_{hy} \Delta
+ * t\right)\right]
+ * + \Sigma(t)\exp\left(-\left(\hat{l}/w\right)^2\alpha_{th} \Delta t\right)\\\end{aligned}\f]
+ * We then compute the pressure and temperature update with an inverse Fourier transform from
+ * \f$\Pi, \Theta\f$.
+ */
+class ThermalPressurization {
+  public:
+  ThermalPressurization(DRParameters& drParameters) : drParameters(drParameters){};
+
+  private:
+  DRParameters& drParameters;
+
+  protected:
+  real (*temperature)[misc::numPaddedPoints];
+  real (*pressure)[misc::numPaddedPoints];
+  real (*theta)[misc::numPaddedPoints][misc::numberOfTPGridPoints];
+  real (*sigma)[misc::numPaddedPoints][misc::numberOfTPGridPoints];
+  real (*thetaTmpBuffer)[misc::numPaddedPoints][misc::numberOfTPGridPoints];
+  real (*sigmaTmpBuffer)[misc::numPaddedPoints][misc::numberOfTPGridPoints];
+  real (*halfWidthShearZone)[misc::numPaddedPoints];
+  real (*hydraulicDiffusivity)[misc::numPaddedPoints];
+  real (*faultStrength)[misc::numPaddedPoints];
+
+  public:
+  /**
```

Comment:
Let's list all public methods on the top, then protected and then private. I personally don't like the following style:
```
class A {
public:
  void foo1();
private:
  int a{};
public:
  void bar1();
protected:
  float b{};
};
```

I usually prefer this one:
```
class A {
public:
  void foo1();
  void bar1();
protected:
  float b{};
private:
  int a{};
};
```

My rationale is the following: we start reading from the top to bottom, and we usually want to know what an object of a class can do (i.e., what is the interface). Then if we want to extend a class we can scroll further and examine what we can/need to override. If we really need to change the class then, yes, we will scroll till the bottom to examine the internals of the class